tempesta_fw/sock_srv.c

```diff
@@ -608,6 +608,11 @@ tfw_srv_conn_alloc(void)
 	INIT_LIST_HEAD(&srv_conn->nip_queue);
 	spin_lock_init(&srv_conn->fwd_qlock);
 
+	if (!(srv_conn->stream.parser.pool = __tfw_pool_new(0))) {
+		kmem_cache_free(tfw_srv_conn_cache, srv_conn);
+		return NULL;
+	}
```

Comment:
This means that we'll use an additional system page for each response, but it's quite unlikely that we'll utilize the whole page just for transfer encoding descriptors.

I think the proper fix for this is would be to:
1. create/allocate `TfwStr` with some initial size
2. fix the TODO for `__str_grow_tree()` and use exponential growth

The initial size for `cut` should be derived from a real life example, e.g. get and average or maximum number of chunks for our web site. Also having that the string descriptors aren't so big, it makes sense to provision a little bit.

Reply:
No, I don't think it's so severe as a page for each response. It's a pool per server connection. It's used when a message is parsed, and then is cleared in `tfw_http_msg_del_parsed_cuts()`. Next message processing reuses the same pool.
And since this pool is used only to store chunks of `parser.cut`, any growth strategy should be fine, so no changes should be required there.